nifi-nar-bundles/nifi-oauth-bundle/nifi-oauth/src/main/java/org/apache/nifi/oauth/AbstractOAuthControllerService.java

```diff
@@ -0,0 +1,172 @@
+                    " response field is different this is where you can change that.")
+            .defaultValue("access_token")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RESPONSE_EXPIRE_TIME_FIELD_NAME = new PropertyDescriptor
+            .Builder().name("JSON response 'expire_time' name")
+            .displayName("JSON response 'expire_time' name")
+            .description("Name of the field in the JSON response that contains the expire time. IETF OAuth2 spec default is 'expire_time' if your API provider's" +
+                    " response field is different this is where you can change that.")
+            .defaultValue("expire_time")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RESPONSE_EXPIRE_IN_FIELD_NAME = new PropertyDescriptor
+            .Builder().name("JSON response 'expire_in' name")
+            .displayName("JSON response 'expire_in' name")
+            .description("Name of the field in the JSON response that contains the expire in. IETF OAuth2 spec default is 'expire_in' if your API provider's" +
+                    " response field is different this is where you can change that.")
+            .defaultValue("expire_in")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RESPONSE_TOKEN_TYPE_FIELD_NAME = new PropertyDescriptor
+            .Builder().name("JSON response 'token_type' name")
+            .displayName("JSON response 'token_type' name")
+            .description("Name of the field in the JSON response that contains the token type. IETF OAuth2 spec default is 'token_type' if your API provider's" +
+                    " response field is different this is where you can change that.")
+            .defaultValue("token_type")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RESPONSE_SCOPE_FIELD_NAME = new PropertyDescriptor
+            .Builder().name("JSON response 'scope' name")
+            .displayName("JSON response 'scope' name")
+            .description("Name of the field in the JSON response that contains the scope. IETF OAuth2 spec default is 'scope' if your API provider's" +
+                    " response field is different this is where you can change that.")
+            .defaultValue("scope")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor EXPIRE_TIME_SAFETY_NET = new PropertyDescriptor
+            .Builder().name("Expire time safety net in seconds")
```

Comment:
The `name` should avoid special characters and spaces for serialization purposes. Suggest `.name("expire_time_safety_net")` (intentionally do not include units in name to allow for future evolution).